### ISSUE TITLE
[TF2] [Vulkan] [HDR] vulkan not playing nice with windows HDR?

### ISSUE THREAD

**Issue by GentlePuppet** [ValveSoftware/Source-1-Games#5918]
When I had Vulkan enabled it didn't play nice with my HDR being enabled on the latest version of windows 10.
Note this doesn't happen with Vulkan is not enabled.

It made the screen darken when I alt-tabbed like it was forcing HDR off and it did the same when I closed the game.
I haven't had any other HDR games do that to me.

The game will blackout my screens and would make the screen darker if alt-tabbed and I would have to turn hdr in both my monitor osd and windows off and on again to restore the normal expected brightness.

I am playing tf2 in borderless windowed and with mat_hdr_level 2 if that matters.

That is the only reason I'm not turning vulkan on all the time. It does make the game feel smoother to play but if it's blacking out my screens anytime I alt-tab and breaking hdr when I close the game it's not worth the trade-off.

It's difficult to really tell the difference from the video but the screen is noticeably darker in-person after closing the game.
Apologies for the quality I had to compress it a bit to upload it here.
Click to watch the video:
https://github.com/ValveSoftware/Source-1-Games/assets/43224790/eb7d3a63-336c-4c44-ba68-273c3b07771e

**Comment by treacherousfiend** [comment]
Potentially worth noting, the HDR in the game has absolutely nothing to do with the HDR on your monitor or on windows. the game does not support HDR rendering, and what the "HDR" setting does is essentially just enable light adaption (you know how you go from a dark area to a lighter area and your eyes have to adjust? its that) and bloom.

**Comment by lalo311** [comment]
Hello, I fixed this issue by following this guide : https://www.reddit.com/r/nvidia/comments/1atc23m/comment/kqwkiqp/, combine that with [ForceAutoHDR](https://github.com/tobeadev/Force_Auto_HDR_GUI) and https://github.com/dylanraga/win11hdr-srgb-to-gamma2.2-icm and you can enjoy TF2 in proper HDR and with the new swap chain so alt tabs aren't a torture.

**Comment by GentlePuppet** [comment]
Yeah that might work for you on 11 but as stated I use windows 10 so autoHDR isn't a thing I can use.

**Comment by lalo311** [comment]
You can still try the swapchain thing

**Comment by megakarlach** [comment]
> Potentially worth noting, the HDR in the game has absolutely nothing to do with the HDR on your monitor or on windows. the game does not support HDR rendering, and what the "HDR" setting does is essentially just enable light adaption (you know how you go from a dark area to a lighter area and your eyes have to adjust? its that) and bloom.

Actually the HDR feature in-game refers to HDR rendering (or "HDR lighting"), and it's more than just light adaption/autoexposure since the game did actually load HDR textures used on skybox but because Source was built for Direct3D 9 and HDR displays wasn't a thing back then, it was tonemapped and downscaled to SDR color output. What Source doesn't support is "HDR display output" or "HDR color".
